### ISSUE TITLE
python2 SyntaxError argument order for save method

### ISSUE THREAD

**Issue by senselessDev** [nschloe/matplotlib2tikz#285]
Python 2.7.6 is failing on import of `matplotlib2tikz` with 

```
  File "/home/xxx/.local/lib/python2.7/site-packages/matplotlib2tikz/save.py", line 237
    def save(filepath, *args, encoding=None, **kwargs):
                                     ^
SyntaxError: invalid syntax
```

Problem is the order of `args`/`kwargs` and keyword argument `encoding`.
Solution would be reordering to `def save(filepath, encoding = 'utf-8', *args, **kwargs):`. 
You can find a discussion about that [here](https://stackoverflow.com/questions/15301999/default-arguments-with-args-and-kwargs) for example.

**Comment by nschloe** [comment]
This messes up the argument order. If someone uses `*args`, it'll get set incorrectly after a change like this.

Better upgrade to Python 3, it's high time to do that anyways.

**Comment by senselessDev** [comment]
> This messes up the argument order. If someone uses `*args`, it'll get set incorrectly after a change like this.
> 
> Better upgrade to Python 3, it's high time to do that anyways.

Well, then I suggest to include [python_requires='>=3.0'](https://www.python.org/dev/peps/pep-0345/#requires-python) in `setup.py` to indicate version requirement on PyPI. It's quite useless to be able to `pip2 install matplotlib2tikz` only to notice that importing this package will result in a SyntaxError.

> Better upgrade to Python 3, it's high time to do that anyways.

Am I allowed to decide on my own when to upgrade? Great, thanks... 
Very unnecessary insult...

**Comment by nschloe** [comment]
> Well, then I suggest to include python_requires='>=3.0' in setup.py to indicate version requirement on PyPI.

Definitely a good idea.

> Am I allowed to decide on my own when to upgrade? Great, thanks... Very unnecessary insult...

Certainly didn't mean to insult, but merely as a good piece of advice from a senior developer. [Python 2 support is running out at the end of this year](https://pythonclock.org/), so all projects that haven't switched over by then will effectively be unsupported. Also, [most big Python projects](https://python3statement.org/) have stopped or will stop supporting Python 2 very soon. I can just reiterate: It's high time to make the switch.

**Comment by PetePupalaikis** [comment]
I just released an open-source software project.  I took the effort to support both python 2 and python 3.  It was not difficult to support both for my project. (It was hard to figure out how to do that, as it originally only ran on Python 2).   When I monitor the download statistics for my project, it looks like about half are running Python 2 and half are running Python 3.  I agree it's probably time for people to upgrade, but only supporting Python 3 will alienate a lot of users who don't agree that it's time to upgrade.
